CflsFromPath must correctly process a case when the path is absent.

`CflsFromPath` fails with the following exception when the provided path leads to nowhere:

```
java.lang.RuntimeException: java.io.FileNotFoundException: /some_unexisting_path (No such file or directory)
	at oo.atom.codegen.bytebuddy.cfls.CflsJar.classFileLocator(CflsJar.java:53)
	at oo.atom.codegen.bytebuddy.cfls.CflsInferred.classFileLocator(CflsInferred.java:43)
	at oo.atom.codegen.bytebuddy.cfls.AssertClassFileLocatorSourceDoesntResolveAClassName.check(AssertClassFileLocatorSourceDoesntResolveAClassName.java:54)
	at oo.atom.tests.TestCase.execute(TestCase.java:53)
	at oo.atom.tests.TestsSuite.lambda$null$0(TestsSuite.java:63)
	at org.junit.jupiter.engine.descriptor.JupiterTestDescriptor.executeAndMaskThrowable(JupiterTestDescriptor.java:141)
	at org.junit.jupiter.engine.descriptor.DynamicTestTestDescriptor.execute(DynamicTestTestDescriptor.java:41)
	at org.junit.jupiter.engine.descriptor.DynamicTestTestDescriptor.execute(DynamicTestTestDescriptor.java:24)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.lambda$executeRecursively$3(HierarchicalTestExecutor.java:113)
	at org.junit.platform.engine.support.hierarchical.SingleTestExecutor.executeSafely(SingleTestExecutor.java:66)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.executeRecursively(HierarchicalTestExecutor.java:108)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.execute(HierarchicalTestExecutor.java:79)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.lambda$executeRecursively$0(HierarchicalTestExecutor.java:115)
	at java.util.Optional.ifPresent(Optional.java:159)
	at org.junit.jupiter.engine.descriptor.TestFactoryTestDescriptor.lambda$invokeTestMethod$1(TestFactoryTestDescriptor.java:92)
	at org.junit.jupiter.engine.execution.ThrowableCollector.execute(ThrowableCollector.java:40)
	at org.junit.jupiter.engine.descriptor.TestFactoryTestDescriptor.invokeTestMethod(TestFactoryTestDescriptor.java:79)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:113)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:58)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.lambda$executeRecursively$3(HierarchicalTestExecutor.java:113)
	at org.junit.platform.engine.support.hierarchical.SingleTestExecutor.executeSafely(SingleTestExecutor.java:66)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.executeRecursively(HierarchicalTestExecutor.java:108)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.execute(HierarchicalTestExecutor.java:79)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.lambda$executeRecursively$2(HierarchicalTestExecutor.java:121)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.Iterator.forEachRemaining(Iterator.java:116)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.lambda$executeRecursively$3(HierarchicalTestExecutor.java:121)
	at org.junit.platform.engine.support.hierarchical.SingleTestExecutor.executeSafely(SingleTestExecutor.java:66)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.executeRecursively(HierarchicalTestExecutor.java:108)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.execute(HierarchicalTestExecutor.java:79)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.lambda$executeRecursively$2(HierarchicalTestExecutor.java:121)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.Iterator.forEachRemaining(Iterator.java:116)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.lambda$executeRecursively$3(HierarchicalTestExecutor.java:121)
	at org.junit.platform.engine.support.hierarchical.SingleTestExecutor.executeSafely(SingleTestExecutor.java:66)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.executeRecursively(HierarchicalTestExecutor.java:108)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor$NodeExecutor.execute(HierarchicalTestExecutor.java:79)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:55)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:43)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:170)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:154)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:90)
	at org.junit.platform.surefire.provider.JUnitPlatformProvider.invokeSingleClass(JUnitPlatformProvider.java:148)
	at org.junit.platform.surefire.provider.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:130)
	at org.junit.platform.surefire.provider.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:109)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:290)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:242)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:121)
Caused by: java.io.FileNotFoundException: /some_unexisting_path (No such file or directory)
	at java.util.zip.ZipFile.open(Native Method)
	at java.util.zip.ZipFile.<init>(ZipFile.java:225)
	at java.util.zip.ZipFile.<init>(ZipFile.java:155)
	at java.util.jar.JarFile.<init>(JarFile.java:166)
	at java.util.jar.JarFile.<init>(JarFile.java:130)
	at net.bytebuddy.dynamic.ClassFileLocator$ForJarFile.of(ClassFileLocator.java:610)
	at oo.atom.codegen.bytebuddy.cfls.CflsJar.classFileLocator(CflsJar.java:51)
	... 63 more
```

It must correctly handle such case and return empty `ClasssFileLocator`.

--- a/atom-basis/src/main/java/oo/atom/codegen/bytebuddy/cfls/CflsFromPath.java
+++ b/atom-basis/src/main/java/oo/atom/codegen/bytebuddy/cfls/CflsFromPath.java
@@ -24,6 +24,8 @@
 
 package oo.atom.codegen.bytebuddy.cfls;
 
+import net.bytebuddy.dynamic.ClassFileLocator;
+
 import java.nio.file.Files;
 import java.nio.file.Path;
 
@@ -41,6 +43,11 @@ class CflsFromPathInference implements ClassFileLocatorSource.Inference {
 
     @Override
     public final ClassFileLocatorSource classFileLocatorSource() {
+        if(!Files.exists(path)) {
+            return new CflsExplicit(
+                ClassFileLocator.NoOp.INSTANCE
+            );
+        }
         if(Files.isDirectory(path)) {
             return new CflsDirectory(path);
         } else {

--- a/atom-basis/src/test/java/oo/atom/codegen/bytebuddy/cfls/CflsFromPathTest.java
+++ b/atom-basis/src/test/java/oo/atom/codegen/bytebuddy/cfls/CflsFromPathTest.java
@@ -1,0 +1,48 @@
+package oo.atom.codegen.bytebuddy.cfls;
+
+import oo.atom.tests.TestCase;
+import oo.atom.tests.TestsSuite;
+
+import java.nio.file.Paths;
+
+class CflsFromPathTest extends TestsSuite {
+    private static final String TEST_CLASS_PATH = System.getProperty("user.dir") + "/target/test-classes";
+
+    static {
+        System.out.println("user.dir = " + TEST_CLASS_PATH);
+    }
+
+    public CflsFromPathTest() {
+        super(
+            new TestCase(
+                "doesn't fail if the path is absent",
+                new AssertClassFileLocatorSourceDoesntResolveAClassName(
+                    new CflsFromPath(
+                        Paths.get("/some_unexisting_path")
+                    ),
+                    "SomeUnexistingClass"
+                )
+            ),
+            new TestCase(
+                "successfully resolves a class from path",
+                new AssertClassFileLocatorSourceResolvesAClassName(
+                    new CflsFromPath(
+                        Paths.get(TEST_CLASS_PATH)
+                    ),
+                    Foo.class.getName()
+                )
+            ),
+            new TestCase(
+                "doesn't resolve absent class from path",
+                new AssertClassFileLocatorSourceDoesntResolveAClassName(
+                    new CflsFromPath(
+                        Paths.get(TEST_CLASS_PATH)
+                    ),
+                    "SomeUnexistingClass"
+                )
+            )
+        );
+    }
+
+    private static final class Foo {}
+}

